### PR TITLE
feat: add `every` method to `array/fixed-endian-factory`

### LINKED ISSUE #3138: [RFC]: add `every` method to `array/fixed-endian-factory`
### Description

This RFC proposes adding support for an `every` method to [`@stdlib/array/fixed-endian-factory`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/fixed-endian-factory).

This method is the equivalent of the [`TypedArray.prototype.every`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/every) method, but for fixed-endian typed arrays.

When adding support, the following tasks should be completed:

1. The `./lib/main.js` file should be updated with the added method.
2. A new test file should be created which includes unit tests testing against expected behavior.
3. Two benchmark files should be added: one measuring baseline performance and another measuring performance as the array length increases.
4. The README should be updated to include documentation for the added method.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/commit/956a4624c788689b1bca285856b987ea3aa32eb6, which is a commit adding `forEach` support. This should provide an idea of what is expected.

For an example of specialized typed arrays having customized methods, see

- [`@stdlib/array/bool`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/bool)
- [`@stdlib/array/complex128`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/complex128)
- [`@stdlib/array/complex64`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/complex64)

While the method proposed in this RFC will likely **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages.

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by aayush0325**
Resolves #3138

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `every` method to `array/fixed-endian-factory` with required docs, tests and benchmarks

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3138 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by aayush0325** [comment]
kindly give this a review @kgryte!

File: lib/node_modules/@stdlib/array/fixed-endian-factory/README.md [hunk 1]
```diff
@@ -338,6 +338,55 @@ v = arr.at( -100 );
...
+};
+
+var bool = arr.every( isPositive, context );
+// returns false
+
+var count = context.count;
+// returns 3
+```
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/benchmark/benchmark.every.length.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+* Predicate function.
+*
+* @private
+* @param {boolean} value - array element
+* @param {NonNegativeInteger} idx - array element index
+* @param {TypedArray} arr - array instance
+* @returns {boolean} boolean indicating whether a value passes a test
+*/
+function predicate( value ) {
+	return value > 0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/benchmark/benchmark.every.length.js]:
```suggestion
	return value >= 0;
```

Given that `zeroTo` returns an array whose first element is `0`, you need to ensure that the predicate function passes for all values in order to actually measure perf as a function of array length.

File: lib/node_modules/@stdlib/array/fixed-endian-factory/benchmark/benchmark.every.length.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	var f;
+	var i;
+
+	min = 1; // 10^min
+	max = 6; // 10^max
+
+	for ( i = min; i <= max; i++ ) {
+		len = pow( 10, i );
+		f = createBenchmark( len );
+		bench( pkg+':forEach:len='+len, f );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/benchmark/benchmark.every.length.js]:
```suggestion
		bench( pkg+':every:len='+len, f );
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js [hunk 4]
```diff
@@ -514,6 +514,37 @@ function factory( dtype ) { // eslint-disable-line max-lines-per-function, stdli
 		return this._buffer.byteLength;
 	});
 
+	/**
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]: This needs to be moved to before `forEach`. Currently, this is not inserted in alphabetical order.

File: lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.every.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.every( value );
+		};
+	}
+});
+
+tape( 'the method returns `true` if operating on an empty boolean array', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.every.js]:
```suggestion
tape( 'the method returns `true` if operating on an empty array', function test( t ) {
```

**Review by kgryte — CHANGES_REQUESTED** [review]
Apart from reordering declared methods, this PR is looking good.

**Comment by aayush0325** [comment]
I've changed the position of `every` method to be before the `forEach` method.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aayush0325!

**Pull request merged**